test_elasticsearch_dsl/test_integration/test_document.py

```diff
@@ -67,7 +69,13 @@ class Meta:
 @fixture
 def pull_request(write_client):
     PullRequest.init()
-    pr = PullRequest(_id=42, comments=[Comment(content='Hello World!', author=User(name='honzakral'))])
+    pr = PullRequest(_id=42,
+                     comments=[
+                         Comment(content='Hello World!',
+                                 author=User(name='honzakral'),
+                                 created_at='2018-01-09T10:17:03.021184'),
```

Comment:
we are now doing deserialization only when loading data form elasticsearch. That means you should be assiging a `datetime` object here. Could you please change this to `datetime(2018, 1, 9, 10, 17, 3, 21184)`?

Thank you!

Reply:
I'm not sure I have completely understood this point after all. Elasticsearch accepts strings as date types. It is `elasticsearch-py` that converts it to a string behind the hood. I'm totally fine with that. But I get confused with the IP type. If I set `comment.ip = '127.0.0.1'` and retrieve this from Elasticsearch, I'll get `comment.ip == ip_address.IPv4Address('127.0.0.1')` when retrieving the document. However, if I try to set `comment.ip = ip_address.IPv4Address('127.0.0.1')`, I'll get a serialization error. Basically, it means that I can't do something like `comment0.ip = comment1.ip`. Is it another issue? Or am I misunderstanding something?

Reply:
Yes, elasticsearch understand strings and you code actually worked without the comment I made, it just wasn't best practice (and people often look into the tests to see those).

I am getting no error when I have an `Ip` field defined on my `DocType` and then assigning `IPv4Address` I can correctly serialize the document (I added a test for it in 7e7c00838e68532522dd386045758de8f839640b). If that doesn't work for you it's definitely a bug and an issue would be great!

Reply:
Uuuuh my bad. Indeed, when I test locally it works. Still, I can reproduce the issue in my app. I have to figure out what causes `to_dict` to not serialize the `IPv4Address`.

Reply:
Got it. The IP field is in an inner object that I set "the old way": using a dictionary and not a proper object (e.g. `PullRequest(comment={'ip': ip_address('127.0.0.1')}])` instead of `PullRequest(comment=Comment(ip=ip_address('127.0.0.1')))`). Don't know if this is supposed to still be supported, but it's way way simpler in my case.